[Identity] Update Readme (#35614)
Azure/azure-sdk-for-js: sdk/identity/identity-broker/README.md

- An [Azure subscription](https://azure.microsoft.com/free/nodejs/).

> > Note: If you are planning to do local development with `@azure/identity-broker` you may need to install some additional tools. [node-gyp](https://github.com/nodejs/node-gyp) is used to compile [addons](https://nodejs.org/api/addons.html) for accessing system APIs. Installation requirements are listed on the [node-gyp README](https://github.com/nodejs/node-gyp#installation)

Reviewer: ```suggestion
> Note: For local development with `@azure/identity-broker`, you may need to install additional tools. [node-gyp](https://github.com/nodejs/node-gyp) is used to compile [addons](https://nodejs.org/api/addons.html) for accessing system APIs. Installation requirements are listed in the [node-gyp README](https://github.com/nodejs/node-gyp#installation).
```

> On Linux, the dependency library uses `libsecret` so you may need to install it. Depending on your distribution, you will need to run the following command:

Reviewer: I'm not sure what's meant by "dependency library" in this context.

Author: It's referring to `@azure/msal-node-extensions` because that's being used by a dependency there. I think we can change it to `the library uses` so we don't have to surface the dependency structure to `keytar`. Would that work?

> On linux, the dependency library uses `libsecret` so you may need to install it. Depending on your distribution, you will need to run the following command:

Reviewer: ```suggestion
On Linux, the dependency library uses `libsecret` so you may need to install it. Depending on your distribution, you will need to run the following command:
```
